frame/staking/src/pallet/impls.rs

```diff
@@ -1629,30 +1626,30 @@ impl<T: Config> StakingInterface for Pallet<T> {
 			.map_err(|with_post| with_post.error)
 	}
 
-	fn bond(
-		who: &Self::AccountId,
-		value: Self::Balance,
-		payee: &Self::AccountId,
-	) -> DispatchResult {
-		Self::bond(
-			RawOrigin::Signed(who.clone()).into(),
-			T::Lookup::unlookup(who.clone()),
-			value,
-			RewardDestination::Account(payee.clone()),
-		)
+	fn desired_validator_count() -> u32 {
+		ValidatorCount::<T>::get()
 	}
 
-	fn nominate(who: &Self::AccountId, targets: Vec<Self::AccountId>) -> DispatchResult {
-		let ctrl = Self::bonded(who).ok_or(Error::<T>::NotStash)?;
-		let targets = targets.into_iter().map(T::Lookup::unlookup).collect::<Vec<_>>();
-		Self::nominate(RawOrigin::Signed(ctrl).into(), targets)
+	fn election_ongoing() -> bool {
+		T::ElectionProvider::ongoing()
 	}
 
-	sp_staking::runtime_benchmarks_enabled! {
-		fn nominations(who: Self::AccountId) -> Option<Vec<T::AccountId>> {
-			Nominators::<T>::get(who).map(|n| n.targets.into_inner())
-		}
+	fn force_unstake(who: Self::AccountId) -> sp_runtime::DispatchResult {
```

Comment:
re-ordering this stuff makes the PR unnecessarily diccicult to review, would be good if you revert those.